### PR TITLE
 BUGFIX: ONE-4067 Move conversion to the AttributeFilter and fix text element values

### PR DESCRIPTION
**Pull request by ivan-nejezchleb**
Improve handling of text filter selection. Move conversion from inner component to the AttributeFilter to improve maintainability

---

[Check PR owner responsibilities](https://confluence.intgdc.com/display/Development/Code-reviews#Code-reviews-Ownerresponsibilities)

Supported PR commands:

Command | Description
--- | ---
`extended test - examples` | Live examples tests
`extended test - storybook` | Storybook screenshot tests

See more [options](https://confluence.intgdc.com/display/kbhr/How+to+work+with+git+and+Github#HowtoworkwithgitandGithub-Extendedchecks).

# PR Checklist

- [ ] Verify code changes ([Checklist](https://confluence.intgdc.com/display/Development/Code-reviews+checklist), [Best practices](https://confluence.intgdc.com/display/Development/Code-reviews+best+practices))
- [ ] [Verify pull-request formalities](https://confluence.intgdc.com/display/Development/Code-reviews)
- [ ] Change was tested by using [gdc-dev-release](https://confluence.intgdc.com/display/~tomas.vojtasek/Private+NPM) in [gdc-analytical-designer](https://github.com/gooddata/gdc-analytical-designer) and [gdc-dashboards](https://github.com/gooddata/gdc-dashboards) (if applicable)
- [ ] Migration guide (for major changes) is mentioned in [CHANGELOG.md](../blob/master/CHANGELOG.md).
- [ ] Successful `extended test - examples`
- [ ] Successful `extended test - storybook`
- [ ] Checked yarn.lock consistency (no dep. duplicities especially Goodstrap)

# Related PRs

- gdc-analytical-designer:
- gdc-dashboards:

# Related Jira tasks

### CODE REVIEW

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/4861/ : SUCCESS in 4m 25s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/166/ : SUCCESS in 6s

**Comment by yenkins** [comment]
:arrow_right: Started Live Examples deploy to:
https://gdc-live-examples-1186.herokuapp.com

It will be destroyed on merge/close. See [ci-infra](https://github.com/gooddata/ci-infra/blob/master/jenkins/jobs/templates/orchestrators/client-pr-dispatcher.groovy) and [gdc-ci.yaml](https://github.com/gooddata/gooddata-react-components/blob/master/gdc-ci.yaml).

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/168/ : SUCCESS in 9s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/1963/ : SUCCESS in 15m 33s

**Comment by ivan-nejezchleb** [comment]
extended test - examples

**Comment by yenkins** [comment]
:arrow_right: Started Live Examples deploy to:
https://gdc-live-examples-1186.herokuapp.com

It will be destroyed on merge/close. See [ci-infra](https://github.com/gooddata/ci-infra/blob/master/jenkins/jobs/templates/orchestrators/client-pr-dispatcher.groovy) and [gdc-ci.yaml](https://github.com/gooddata/gooddata-react-components/blob/master/gdc-ci.yaml).

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/4867/ : SUCCESS in 9m 51s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/175/ : SUCCESS in 14s

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/176/ : SUCCESS in 13s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/1967/ : SUCCESS in 19m 41s

File: src/components/filters/AttributeFilter/AttributeFilter.tsx [hunk 1]
```diff
@@ -111,6 +145,19 @@ export class AttributeFilter extends React.PureComponent<IAttributeFilterProps>
...
     }
 
+    public onApply = (selection: IAttributeElement[], isInverted: boolean) => {
+        const { identifier, uri } = this.getIdentifierOrUri();
+        let isUsingTextFilter = false;
+        if (this.props.filter) {
+            const filterBody = this.getFilterBody(this.props.filter);
+            isUsingTextFilter = Boolean(filterBody.textFilter);
+        }
+        const id: string = identifier || uri;
```
[the remark this answers is not in the document]
Comment (ivan-nejezchleb) [reply src/components/filters/AttributeFilter/AttributeFilter.tsx]: This code do the same as code in https://github.com/gooddata/gooddata-react-components/blob/next-7.2.0/src/components/filters/AttributeFilter/AttributeLoader.tsx#L128 which provided `isUsingIdentifier` and `attributeDisplayForm` in original implementation. I can re-write it into 

`const isUsingIdentifier = !!identifier;
const id: string = isUsingIdentifier ? identifier : uri;`

to be more similar, but result should be the same

[the remark this answers is not in the document]
Comment (pbenes) [reply src/components/filters/AttributeFilter/AttributeFilter.tsx]: Keep your version, just wanted to confirm, thanks a lot for pointing me to the right place. Closing the remark.

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/2566/ : SUCCESS in 10m 16s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/4872/ : SUCCESS in 5m 13s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-merge-compliance-check/1221/ : SUCCESS in 15s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-examples-zuul-docker/1599/ : SUCCESS in 9m 51s

**Pull request merged**